### ISSUE TITLE
DNS redirect iptables rules not created by istio-cni

### ISSUE THREAD

**Issue by SheilaghM** [rancher/rancher#36159]
**Rancher Server Setup**
- Rancher version:  v2.5.8
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): Istio 1.8
- Proxy/Cert Details:

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

   
Description
Issue description:
DNS redirect iptables rules are not created by istio-cni as a result of upstream issue tracked in https://github.com/istio/istio/issues/29511 This appears to be targetted for fix in istio 1.11 per https://github.com/istio/istio/pull/32416#issuecomment-862625770 so user request is to have istio 1.11 available via Rancher as soon as possible following upstream release

Business impact:
As of Istio 1.8 use of a the .global stub domain for multi-primary cluster istio configurations is not recommended and affected by several bug reports

Troubleshooting steps:
N/A

Repro steps:

See upstream issue

Actual behavior:
DNS redirect iptables rules not created by istio-cni

Expected behavior:
DNS redirect iptables rules successfully created by istio-cni

Files, logs, traces:
N/A

Additional notes:
N/A

Options

**Comment by SheilaghM** [comment]
Duplicate of #29511 - which is already closed.

**Comment by rayandas** [comment]
We validated this on Rancher v2.6.3 with Istio 1.11.4 and we can confirm that this issue is getting fixed on Istio 1.11.4
Steps taken to validate:
1. Install Rancher v2.6.3
2. Install rancher-monitoring:100.1.0+up19.0.3
3. Install rancher-istio:100.1.0+up1.11.4  with Istio operator mentioned in upstream [issue](https://github.com/istio/istio/issues/29511#:~:text=Install%20istio%20with%20cni.%20We%20used%20the%20following%20istio%20operator%20file%20%3A%2D)
4. Enable Auto Injection in default namespace
5. Create a sample [service](https://github.com/istio/istio/blob/master/samples/httpbin/httpbin.yaml) `httpbin`
6. Create a [Service Entry](https://github.com/istio/istio/issues/29511#:~:text=Create%20a%20service%20entry.%20We%20used%20following%20%3A%2D)
7. SSH into Host machine
8.On host machine get the PID of the `httpbin` service created in Step 5
9. Get into the namespace of `httpbin` using command : `nsenter --target <PID> --uts --ipc --net --pid`
Verify the iptables rules created by istio-cni plugin has UDP rules to redirect the dns lookup to the sidecar, command: `iptables -t nat --list | grep -i udp`
Expected Output:
```
Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
ISTIO_OUTPUT  tcp  --  anywhere             anywhere            
RETURN     udp  --  anywhere             anywhere             udp dpt:domain owner UID match 1337
RETURN     udp  --  anywhere             anywhere             udp dpt:domain owner GID match 1337
REDIRECT   udp  --  anywhere             anywhere             udp dpt:domain redir ports 15053 
```
9. As the `httpbin` service is not exposed outside of the cluster qw cannot curl it directly, however we can verify that it is working correctly using a curl command against `httpbin:8000` from inside the cluster using the public `dockerqa/curl` image from Docker hub: Commands used 
```
kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/html
kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent --head httpbin:8000/status/500
time kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/delay/5
```
Also we tested the httpbin service by starting the [sleep service](https://github.com/istio/istio/blob/master/samples/sleep) and calling httpbin from it. Steps:
1. Start the sleep service using command
```
kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/sleep/sleep.yaml 
```
2. Now `kubectl exec` into the sleep service to experiment with Istio networking. The following command can be used to call the `httpbin` service.
```
export SLEEP_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})          
kubectl exec -it $SLEEP_POD -c sleep -- curl http://httpbin:8000/html
```

> We already have PRs to update Istio to `1.11.7` in both 2.5 https://github.com/rancher/charts/pull/1770 & 2.6 https://github.com/rancher/charts/pull/1765, this issue will be fixed there.

**Comment by rishabhmsra** [comment]
Validated on rancher v.2.6-head(f2a4147), docker install. Using the steps mentioned [here](https://github.com/rancher/rancher/issues/36159#issuecomment-1057135734).

**Validation steps followed:**
- Created a k8s `1.22.6-rancher1-2` cluster -> ec2 node driver(1-cp, 1-etcd, 2-worker)
- Installed monitoring chart `100.1.1+up19.0.3`.
- Installed Istio `100.1.1+up1.11.7`, with CNI enabled and Istio operator mentioned [here](https://github.com/istio/istio/issues/29511#:~:text=Install%20istio%20with%20cni.%20We%20used%20the%20following%20istio%20operator%20file%20%3A%2D).
- Enabled Auto Injection in default namespace
- Created a sample [service](https://github.com/istio/istio/blob/master/samples/httpbin/httpbin.yaml) httpbin
- Created a [ServiceEntry](https://github.com/istio/istio/issues/29511#:~:text=Create%20a%20service%20entry.%20We%20used%20following%20%3A%2D). Edited the endpoint to `httpbin.default.svc.cluster.local`
- SSH'ed into worker node and fetched the PID of httpbin service
- Get into the namespace of `httpbin` using command : `nsenter --target <PID> --uts --ipc --net --pid`
- Ran `iptables -t nat --list | grep -i udp` to verify the UDP rules created in iptables :
```
$ nsenter --target 326072 --uts --ipc --net --pid
root@httpbin-74fb669cc6-74zrb:/home/ubuntu# iptables -t nat --list
...
Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
ISTIO_OUTPUT  tcp  --  anywhere             anywhere            
RETURN     udp  --  anywhere             anywhere             udp dpt:domain owner UID match 1337
RETURN     udp  --  anywhere             anywhere             udp dpt:domain owner GID match 1337
REDIRECT   udp  --  anywhere             anywhere             udp dpt:domain redir ports 15053
...
```
- Verified the httpbin service as mentioned [here](https://github.com/rancher/rancher/issues/36159#issuecomment-1057135734) in point no. 9 :
```
kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/html ----> Got the html response
```
```
kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent --head httpbin:8000/status/500
If you don't see a command prompt, try pressing enter.
HTTP/1.1 500 Internal Server Error
server: envoy
date: Tue, 08 Mar 2022 13:05:57 GMT
content-type: text/html; charset=utf-8
access-control-allow-origin: *
access-control-allow-credentials: true
content-length: 0
x-envoy-upstream-service-time: 14
```
```
time kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/delay/5
If you don't see a command prompt, try pressing enter.
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept": "*/*", 
    "Host": "httpbin:8000", 
    "User-Agent": "curl/7.35.0", 
    "X-B3-Parentspanid": "920e766319d4d7de", 
    "X-B3-Sampled": "1", 
    "X-B3-Spanid": "20addcff684885b5", 
    "X-B3-Traceid": "01bf5da95aedccc8920e766319d4d7de", 
    "X-Envoy-Attempt-Count": "1", 
    "X-Forwarded-Client-Cert": "By=spiffe://cluster.local/ns/default/sa/httpbin;Hash=ebbeaee0eaec172ad624189b8e30fd278d6c45ef456fae7a99d46dafa787eaa3;Subject=\"\";URI=spiffe://cluster.local/ns/default/sa/default"
  }, 
  "origin": "127.0.0.6", 
  "url": "http://httpbin:8000/delay/5"
}
```
- Ran below command and getting the html response :
```
export SLEEP_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})          
kubectl exec -it $SLEEP_POD -c sleep -- curl http://httpbin:8000/html
```

- Also accessed the httpbin service through `host` defined in the ServiceEntry and getting the html response:
```
kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin.test.global:8000/html
If you don't see a command prompt, try pressing enter.
<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
      <h1>Herman Melville - Moby-Dick</h1>

      <div>
        <p>
          Availing himself of the mild, summer-cool weather that now reigned
...
```
**Result:**
- The iptables rules created by istio-cni plugin has UDP rules. Also the access to httbin service is working as expected, hence closing the issue.
